### ISSUE TITLE
Type helpers assigned to ipfs-http-client constructor

### ISSUE THREAD

**Issue by dapplion** [ipfs/js-ipfs#3450]
**Expected**
The utilities assigned to the constructor should be typed as existing
```ts
Object.assign(ipfsClient, { CID, multiaddr, multibase, multicodec, multihash, globSource, urlSource })
```
**Current**
ipfsClient type does not include above utilities

**Comment by achingbrain** [comment]
Please can you try with the latest RC?  `ipfs-http-client` has switched to named exports only so the utilities are no longer properties of the factory function:

```console
$ npm install ipfs-http-client@next
```

```js
const {
  create,
  CID,
  multiaddr
  // ..etc
} = require('ipfs-http-client')

const client = create()

console.info(await client.id())
// ..etc
```

**Comment by dapplion** [comment]
Can I install this version's next without getting the following versions next if the package.json is left unchanged for a while?

**Comment by achingbrain** [comment]
Not if you use `next` as the version number, but if you use the current RC version directly you'll aways get that version:

```console
$ npm install ipfs-http-client@49.0.5-rc.3
```

Of course, if there's a subsequent RC you'll miss out on the changes.

**Comment by dapplion** [comment]
@achingbrain Got
```
error ipfs-unixfs@4.0.1: The engine "node" is incompatible with this module. Expected version ">=14.0.0". Got "12.21.0"
error Found incompatible module.
```
You don't support node 12 anymore?

**Comment by achingbrain** [comment]
> You don't support node 12 anymore?

No, only Current and Active LTS - https://github.com/ipfs/community/blob/master/CONTRIBUTING_JS.md#supported-versions.  Node 12 became Maintenance LTS at the end of last year.

**Comment by dapplion** [comment]
> No, only Current and Active LTS - https://github.com/ipfs/community/blob/master/CONTRIBUTING_JS.md#supported-versions. Node 12 became Maintenance LTS at the end of last year.

If possible do you know the last `ipfs-http-client` version that accepted node 12?

**Comment by achingbrain** [comment]
The `engines.node` field of the current release (49.0.4) is `>=10.3.0` so yarn will not error out if you try to install it on node 12.

It's possible you could transpile `@next` to be compatible with node 12 but if possible you should just upgrade.

**Comment by dapplion** [comment]
> The engines.node field of the current release (49.0.4) is >=10.3.0 so yarn will not error out if you try to install it on node 12.

Right, probably the issue is that the various dependencies are not synced on the node version required. In my case it was `ipfs-unixfs@4.0.1` that blocked node 12

**Comment by whizzzkid** [comment]
js-ipfs is being deprecated in favor of [Helia](https://github.com/ipfs/helia). You can follow the migration plan here https://github.com/ipfs/js-ipfs/issues/4336 and [read the migration guide](https://github.com/ipfs/helia/wiki/%F0%9F%9A%9B-Migrating-from-js-IPFS).


This issue has been resolved in Helia! if this does not address your concern please let us know  by reopening this issue before 2023-06-05!
